### PR TITLE
Add the backchannel logout event

### PR DESCRIPTION
**Pull request by 2403905**
## Description

Added SSE-Event when logged out

## Related Issue

- Fixes [#9355](https://github.com/owncloud/ocis/issues/9355)

## How Has This Been Tested?

- test environment: build an image and run [ocis_keycloak](https://github.com/owncloud/ocis/tree/master/deployments/examples/ocis_keycloak)
- test case 1: Login as admin to ocis.owncloud.test, and open one more tab in a browser. Watching the `sse` in a browser console. After logout, the browser has get an event.
The message format: type: `backchannel-logout` data:`{"userid":"93277962-0054-47d5-9717-fed84f26b9e7","timestamp":"seconds:1718982662 nanos:826943110 "}`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: services/clientlog/pkg/service/events.go [hunk 1]
```diff
@@ -11,3 +11,9 @@ type FileEvent struct {
 	// Only in case of sharing (refactor this into separate struct when more fields are needed)
 	AffectedUserIDs []string `json:"affecteduserids"`
 }
+
+// BackchannalLogout is emitted when a user logs in
```
Comment (kobergj) [inline services/clientlog/pkg/service/events.go:15]:
```suggestion
// BackchannelLogout is emitted when a user logs out
```

File: services/clientlog/pkg/service/events.go [hunk 2]
```diff
@@ -11,3 +11,9 @@ type FileEvent struct {
...
+type BackchannalLogout struct {
```
Comment (kobergj) [inline services/clientlog/pkg/service/events.go:16]:
```suggestion
type BackchannelLogout struct {
```

File: services/proxy/pkg/staticroutes/backchannellogout.go [hunk 3]
```diff
@@ -62,3 +71,42 @@ func (s *StaticRouteHandler) backchannelLogout(w http.ResponseWriter, r *http.Re
 	render.Status(r, http.StatusOK)
 	render.JSON(w, r, nil)
 }
+
+// publishBackchannalLogoutEvent publishes a backchannel logout event when a user logs out
+func (s StaticRouteHandler) publishBackchannalLogoutEvent(ctx context.Context, record *microstore.Record, logoutToken *oidc.LogoutToken) error {
```
Comment (kobergj) [inline services/proxy/pkg/staticroutes/backchannellogout.go:76]:
```suggestion
func (s StaticRouteHandler) publishBackchannelLogoutEvent(ctx context.Context, record *microstore.Record, logoutToken *oidc.LogoutToken) error {
```

File: services/proxy/pkg/staticroutes/backchannellogout.go [hunk 4]
```diff
@@ -62,3 +71,42 @@ func (s *StaticRouteHandler) backchannelLogout(w http.ResponseWriter, r *http.Re
 	render.Status(r, http.StatusOK)
 	render.JSON(w, r, nil)
 }
+
+// publishBackchannalLogoutEvent publishes a backchannel logout event when a user logs out
```
Comment (kobergj) [inline services/proxy/pkg/staticroutes/backchannellogout.go:75]:
```suggestion
// publishBackchannelLogoutEvent publishes a backchannel logout event when a user logs out
```

File: services/proxy/pkg/staticroutes/backchannellogout.go [hunk 5]
```diff
@@ -62,3 +71,42 @@ func (s *StaticRouteHandler) backchannelLogout(w http.ResponseWriter, r *http.Re
...
+		return fmt.Errorf("could not get user by claims: %w", err)
+	}
+
+	e := events.BackchannelLogout{
+		Executant: user.GetId(),
+		SessionId: logoutToken.SessionId,
+		Timestamp: utils.TSNow(),
+	}
+
+	if s.EventsPublisher != nil {
```
Comment (kobergj) [inline services/proxy/pkg/staticroutes/backchannellogout.go:106]: This is no good check. We don't need to do anything in this function if we don't have an `EventsPublisher`. Either check it at the beginning of the function and return directly OR check it during initialization and assume it is `!= nil` here. I'd go with second solution, but first one also fine if the publisher is optional.

**Review by kobergj — CHANGES_REQUESTED** [review]

File: services/proxy/pkg/staticroutes/backchannellogout.go (hunk 5 above)
Comment (2403905) [reply services/proxy/pkg/staticroutes/backchannellogout.go]: Do you mean during the proxy service initialization?
If yes I don't think this is a good idea. Sounds to me like we should not interrupt the proxy launching

File: services/proxy/pkg/config/config.go [hunk 6]
```diff
@@ -217,3 +218,14 @@ type ServiceAccount struct {
 	ServiceAccountID     string `yaml:"service_account_id" env:"OCIS_SERVICE_ACCOUNT_ID;PROXY_SERVICE_ACCOUNT_ID" desc:"The ID of the service account the service should use. See the 'auth-service' service description for more details." introductionVersion:"5.0"`
 	ServiceAccountSecret string `yaml:"service_account_secret" env:"OCIS_SERVICE_ACCOUNT_SECRET;PROXY_SERVICE_ACCOUNT_SECRET" desc:"The service account secret." introductionVersion:"5.0"`
 }
+
+// Events combines the configuration options for the event bus.
+type Events struct {
```
Comment (micbar) [inline services/proxy/pkg/config/config.go:223]: Please set the correct `introductionVersion`

**Pull request merged**